src/tools/kata-ctl/src/ops/env_ops.rs

```diff
@@ -0,0 +1,468 @@
+    let (host_distro_name, host_distro_version) =
+        utils::get_distro_details(utils::OS_RELEASE, utils::OS_RELEASE_CLR)?;
+    let (cpu_vendor, cpu_model) = utils::get_generic_cpu_details(check::PROC_CPUINFO)?;
+
+    let host_distro = DistroInfo {
+        name: host_distro_name,
+        version: host_distro_version,
+    };
+
+    let cores: usize = std::thread::available_parallelism()
+        .context("get available parallelism")?
+        .into();
+
+    let host_cpu = CPUInfo {
+        vendor: cpu_vendor,
+        model: cpu_model,
+        cpus: cores,
+    };
+
+    let memory_info = get_memory_info()?;
+
+    let guest_protection = arch_specific::available_guest_protection()?.to_string();
+
+    let support_vsocks = utils::supports_vsocks(utils::VHOST_VSOCK_DEVICE)?;
+
+    Ok(HostInfo {
+        kernel: host_kernel_version,
+        architecture: String::from(std::env::consts::ARCH),
+        distro: host_distro,
+        cpu: host_cpu,
+        memory: memory_info,
+        available_guest_protection: guest_protection,
+        vm_container_capable: true, // TODO: implement
+        support_vsocks,
+    })
+}
+
+pub fn get_runtime_info(toml_config: &TomlConfig) -> RuntimeInfo {
+    let version = VersionInfo {
+        semver: String::from(version::VERSION),
+        commit: String::from(version::COMMIT),
+    };
+
+    let config_path = TomlConfig::get_default_config_file();
+    let mut toml_path = String::new();
+    if config_path.is_ok() {
+        let p = config_path.unwrap();
+        let path_str = p.to_str();
+        toml_path = match path_str {
+            Some(s) => String::from(s),
+            None => String::new(),
+        };
+    }
+
+    RuntimeInfo {
+        // TODO: Needs to be implemented
+        path: "TBD".to_string(),
+        version,
+        experimental: toml_config.runtime.experimental.clone(),
+        // TODO: See https://github.com/kata-containers/kata-containers/issues/6667
+        guest_selinux_label: String::from("not implemented yet"),
+        debug: toml_config.runtime.debug,
+        trace: toml_config.runtime.enable_tracing,
+        disable_guest_seccomp: toml_config.runtime.disable_guest_seccomp,
+        disable_new_net_ns: toml_config.runtime.disable_new_netns,
+        sandbox_cgroup_only: toml_config.runtime.sandbox_cgroup_only,
+        static_sandbox_resource_mgmt: toml_config.runtime.static_sandbox_resource_mgmt,
+        config: RuntimeConfigInfo { path: toml_path },
+    }
+}
+
+pub fn get_agent_info(toml_config: &TomlConfig) -> Result<AgentInfo> {
+    if toml_config.agent.len() == 0 {
+        return Err(anyhow!("Missing agent config"));
+    }
+
+    // Assign the first entry to the agent config after checking hashmap is not empty, to make this
+    // work for configs where agent_name is absent.
+    // This is a workaround for https://github.com/kata-containers/kata-containers/issues/5954
+    let (_, mut agent_config) = toml_config.agent.iter().next().unwrap();
```

Comment:
Since the function returns a `Result`, can you replace the `unwrap()`. with `?` here?

Reply:
The unwrap was for getting the value from Option fetched from next(). I had made sure this does not panic after checking for the length of the hashmap itself.
I'll rewrite this to avoid using unwrap() altogether.